soroban-env-host/src/host.rs

```diff
@@ -2400,4 +2400,19 @@ impl VmCallerCheckedEnv for Host {
         let res = HostVec::from_vec(self.0.budget.clone(), outer)?;
         Ok(self.add_host_object(res)?.into())
     }
+
+    fn fail_with_status(
+        &self,
+        vmcaller: &mut VmCaller<Self::VmUserState>,
+        status: Status,
+    ) -> Result<RawVal, Self::Error> {
+        if status.is_type(ScStatusType::ContractError) {
+            Err(self.err_status_msg(status, "failing with contract-provided status code"))
```

Comment:
@graydon If the currently executing contract is being invoked by another contract, will that other contract get the status as the result of the invocation? Or will the entire stack of invocations be aborted?

Reply:
If A calls B using call, and B fails, A will fail.

If A calls B using try_call, and B fails, A will get the status result.

This extrapolates to the call chain. All calls-of-calls will fail unless there's a try_call somewhere along the chain.

Reply:
Makes sense. This made me realize we do not actually have try_call wired up in the SDK or in the generated clients. I'm going to go plumb that through.